spring-ai-vector-store/src/main/java/org/springframework/ai/vectorstore/SearchRequest.java

```diff
@@ -283,10 +285,22 @@ public Builder filterExpression(@Nullable Filter.Expression expression) {
 		 */
 		public Builder filterExpression(@Nullable String textExpression) {
 			this.searchRequest.filterExpression = (textExpression != null)
-					? new FilterExpressionTextParser().parse(textExpression) : null;
+					? new FilterExpressionTextParser().parse(escapeTextExpression(textExpression)) : null;
 			return this;
 		}
 
+		private String escapeTextExpression(String expression) {
+			Pattern pattern = Pattern.compile("'([^']*)'");
+			Matcher matcher = pattern.matcher(expression);
+			StringBuffer sb = new StringBuffer();
+			while (matcher.find()) {
+				String content = matcher.group(1).replace("\\", "\\\\").replace(".", "\\.");
```

Comment:
```
static Map<String, String> escapeText = {
  // from, to
  "\\", "\\\\",
  ".", "\\."
}
```
nit. how about extracting escape target texts to variable?
maybe we need more escape text pattern in the future.